Rubberduck.Core/UI/UnitTesting/TestExplorerModel.cs

```diff
@@ -125,13 +86,198 @@ public bool IsBusy
             }
         }
 
+        private bool _isRefreshing;
+        public bool IsRefreshing
+        {
+            get => _isRefreshing;
+            set
+            {
+                _isRefreshing = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public string LastTestRunSummary =>
+            string.Format(Resources.UnitTesting.TestExplorer.TestOutcome_RunSummaryFormat, CurrentRunTestCount, Tests.Count, TimeSpan.FromMilliseconds(TotalDuration));
+
+        public int LastTestFailedCount => Tests.Count(test => test.Result.Outcome == TestOutcome.Failed && _testEngine.LastRunTests.Contains(test.Method));
+
+        public int LastTestInconclusiveCount => Tests.Count(test => test.Result.Outcome == TestOutcome.Inconclusive && _testEngine.LastRunTests.Contains(test.Method));
+
+        public int LastTestIgnoredCount => Tests.Count(test => test.Result.Outcome == TestOutcome.Ignored && _testEngine.LastRunTests.Contains(test.Method));
+
+        public int LastTestSucceededCount => Tests.Count(test => test.Result.Outcome == TestOutcome.Succeeded && _testEngine.LastRunTests.Contains(test.Method));
+
+        public void ExecuteTests(IReadOnlyCollection<TestMethodViewModel> tests)
+        {
+            if (!tests.Any())
+            {
+                return;
+            }
+
+            IsBusy = true;
+
+            try
+            {
+                _testEngine.Run(tests.Select(test => test.Method));
+            }
+            catch (Exception ex)
```

Comment:
Would the cancellation get propagated to here? If so, shouldn't we have a `ThreadAbortException` catch block?

Reply:
The cancellation won't raise a `ThreadAbortException` - the thought of mixing a task cancellation on the UI thread with our own custom message pumping tossed in the mix made me cringe.  It's handled by a simple state flag that just lets the invoke exit normally. Pumping the messages lets the cancel button call the `RequestCancellation()` method directly on the `TestEngine`. The initial discussion on the issue #3158 was **_way_** overthinking this.